drivers/hmis/app/graphql/mutations/create_scan_card_code.rb

```diff
@@ -0,0 +1,24 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class CreateScanCardCode < CleanBaseMutation
+    argument :client_id, ID, required: true
+
+    field :scan_card_code, Types::HmisSchema::ScanCardCode, null: true
+
+    def resolve(client_id:)
+      client = Hmis::Hud::Client.viewable_by(current_user).find(client_id)
+      raise 'unauthorized' unless current_permission?(permission: :can_manage_scan_cards, entity: client)
+
+      scan_card_code = Hmis::ScanCardCode.new(client: client, created_by: current_user)
+      scan_card_code.assign_code
+      scan_card_code.save!
```

Comment:
This is totally fine as is, but you could probably just do:
```suggestion
      scan_card_code = Hmis::ScanCardCode.create!(client: client, created_by: current_user)
```
if you move `assign_code` to an after initialize hook.